cmd/libsnap-confine-private/mount-opt.c

```diff
@@ -105,11 +105,11 @@ const char *sc_mount_opt2str(unsigned long flags)
 	// Render any flags that are unaccounted for.
 	if (flags) {
 		char of[128];
-		sprintf(of, "%#lx", flags);
-		strcat(buf, of);
+		sc_must_snprintf(of, sizeof of, "%#lx", flags);
+		sc_string_append(buf, buf_size, of);
 	}
 	// Chop the excess comma from the end.
-	size_t len = strlen(buf);
+	size_t len = strnlen(buf, buf_size);
 	if (len > 0 && buf[len - 1] == ',') {
```

Comment:
Thanks for the tip. I don't have the instinct developed for using `strnlen` (imagine I worked me entire life without knowing it exited). I'll definitely be on the lookout for this more.